roles/README.adoc

```diff
@@ -99,23 +72,15 @@ Moreover, after a major upgrade of their systems, it may force the users to chan
 Exporting `$ROLENAME_provider_os_default` allows the users to set `$ROLENAME_provider: "{{ $ROLENAME_provider_os_default }}"` (thanks to the lazy variable evaluation in Ansible) and thus get a consistent setting for all the systems of the given OS version without having to decide what the actual value is - the decision is delegated to the role).
 ====
 
-== Implementation considerations
-
-=== Role Structure
+=== Distributions and Versions
 [%collapsible]
 ====
-Avoid testing for distribution and version in tasks.
-Rather add a variable file to "vars/" for each supported distribution and version with the variables that need to change according to the distribution and version.
-This way it is easy to add support to a new distribution by
-simply dropping a new file in to "vars/", see below
-<<supporting-multiple-distributions-and-versions,Supporting multiple distributions and versions>>.
-See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding large lists or 'magic values' directly into the playbook."
-Since distribution-specific values are kind of "magic values", it applies to them.
-The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed.
-Consider making paths to templates internal variables if you need different templates for different distributions.
+Explanations:: Avoid testing for distribution and version in tasks. Rather add a variable file to "vars/" for each supported distribution and version with the variables that need to change according to the distribution and version. 
+
+Rationale:: This way it is easy to add support to a new distribution by simply dropping a new file in to "vars/", see below <<supporting-multiple-distributions-and-versions,Supporting multiple distributions and versions>>. See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding large lists or 'magic values' directly into the playbook." Since distribution-specific values are kind of "magic values", it applies to them. The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed. Consider making paths to templates internal variables if you need different templates for different distributions.
```

Comment:
```suggestion
Rationale:: This way it is easy to add support to a new distribution by simply dropping a new file in to "vars/", see below <<supporting-multiple-distributions-and-versions,Supporting multiple distributions and versions>>.
See also <<vars-vs-defaults,Vars vs Defaults>> which mandates "Avoid embedding large lists or 'magic values' directly into the playbook."
Since distribution-specific values are kind of "magic values", it applies to them.
The same logic applies for providers: a role can load a provider-specific variable file, include a provider-specific task file, or both, as needed.
Consider making paths to templates internal variables if you need different templates for different distributions.
```